### ISSUE TITLE
Major issues 

### ISSUE THREAD

**Issue by ghost** [Adamantcheese/Kuroba#359]
You chose the wrong app to fork but I'll help you out.

Add the ability to select images in the gallery viewer and download it.
Fix the loading issues when refreshing boards etc.
Fix the loading issue when refreshing the thread.
When you're at the bottom of a thread or at the last thread in a board, swiping down should refresh the thread/board. Clicking at the thread details on the bottom is inconvenient. This isn't 2010 anymore.
Swiping up or down while searching should refresh the board or thread too.
It takes forever for the scrollbar to go away which is frustrating because it blocks those three dots in a post, making it impossible to quote somebody while quickly scrolling a thread.
Make it easier to scroll with the scrollbar because the bar is too thin.
Add support for flags on /pol/ .
Add some space between each post (2-3 px).
When in a thread, replace the bookmark button with the reply button.
When you want to hide a post, it says "Remove". It should be "Hide".
Fix external apps. When I click on a link, it should open a tab in my browser, not inside the app's internal browser. When I click on a YouTube link, it should open Newpipe and not the default YouTube app. 
When I'm on X's board, and I click on a favorited thread that's on Y board, and I go back, it should go to Y's board, instead of X's board.

can't think of anything else

**Comment by ownBard** [comment]
Half of this isn't even true.

**Comment by ghost** [comment]
Dispute it then, dipshit

**Comment by Adamantcheese** [comment]
Alright, let's go through all these.
1) The album viewer covers this.
2) There are no loading issues that I am aware of. Too vague to fix.
3) There are no loading issues that I am aware of. Too vague to fix.
4) Swiping down scrolls the thread up, I'm not overriding that behavior.
5) This is intentional as it prevents a number of bugs with loading and searching at the same time.
6) The button is large enough that the scrollbar does not cover it. Not an issue.
7) The scrollbar is meant more as an indicator of thread progress. This will not change; also if this did occur, it would make 6 even more of a problem, and it would look really ugly.
8) Flags are already supported.
9) The divider between posts is 1px already. This is sufficient.
10) The reply button is fine where it is. You are free to make your own fork to change this if you want.
11) Removing and hiding a post are different actions. Won't fix.
12) If the external app that opens up the link makes an inset browser, that's on the browser's implementation, not mine. If it's opening in Youtube and not Newpipe, you have your default application set incorrectly.
13) There is nothing preventing you from tapping twice to get to the board you want. Bookmarks are not meant to navigate you away from your last catalog in case you want to go back to it after checking another thread quickly.

tl:dr; [Everything you know is wrong.](https://www.youtube.com/watch?v=KThlYHfIVa8)

**Comment by ghost** [comment]
> The album viewer covers this.

It does, but the implementation is the most retarded thing I've ever seen. 
Any normal gallery viewer = click on the gallery viewer, keep the finger on the image's thumbnail for 2 milliseconds, select other images (the option to select all exists too), click download, done.
Clover/Kuroba's gallery viewer = click on the gallery viewer, click on the little three dots on the upper right corner, click on "download album", unselect everything because it will automatically select every image by default for whatever retarded reason, select the images you want to download in a bulk, click download, done

> There are no loading issues that I am aware of. Too vague to fix.
> There are no loading issues that I am aware of. Too vague to fix.

They are well-known issues and they need to be fixed. 

> Swiping down scrolls the thread up, I'm not overriding that behavior.

I meant swiping up and my point still stands. If swiping down will refresh the thread, then swiping up should do the same thing.

> This is intentional as it prevents a number of bugs with loading and searching at the same time.
Every other imageboard app is doing it perfectly fine without any issues. 

> The button is large enough that the scrollbar does not cover it. Not an issue.

No, it isn't. Even with Samsung Note's pen, the button is still too thin and the scrollbar blocks the button.

> The scrollbar is meant more as an indicator of thread progress. This will not change; also if this did occur, it would make 6 even more of a problem, and it would look really ugly.
You say a slightly larger scrollbar will "look really ugly" when there's that jumbo garbage to the right that's cover 1/5 of the screen? I don't think so. A slightly larger scrollbar will be beneficial.

> Flags are already supported.

The option to use flags on /pol/ and other imageboards with meme flags doesn't exist at all.

> The divider between posts is 1px already. This is sufficient.

Making it 2 or 3 px will make it even more sufficient. Especially for the theme "Tomorrow".

> The reply button is fine where it is. You are free to make your own fork to change this if you want.

Clover/Kuroba is the only imageboard app with the reply button inside the menu instead of where it should be. And the bookmark button is confusing as hell because the icon is just a rectangle. Since when did a rectangle represent bookmarks? Usually it's a star.

> Removing and hiding a post are different actions. Won't fix.

Different actions? They're the same. Have you even used Clover or your own fork? Removing a post will literally hide it. It should be changed to "hide" because I can't think of a single imageeboard using the word "remove" to hide a post because it would conflict with deleting. https://i.imgur.com/9LEAaQ8.png

> If the external app that opens up the link makes an inset browser, that's on the browser's implementation, not mine. If it's opening in Youtube and not Newpipe, you have your default application set incorrectly.

No, my default Youtube app is set to Newpipe and every other app is opening it just fine. Clover/Kuroba is the only app opening YouTube links with the default YouTube apps. I've got two Android phones and I managed to replicate this issue (and all of the other issues).

> There is nothing preventing you from tapping twice to get to the board you want. Bookmarks are not meant to navigate you away from your last catalog in case you want to go back to it after checking another thread quickly.

Why should I tap twice when the intended behavior should take me to the thread's board. This is like imageboard 101.
